### ISSUE TITLE
Unit testing a component throws "Required context `history` was not specified"

### ISSUE THREAD

**Issue by thom-nic** [react-bootstrap/react-router-bootstrap#130]
I have a karma+phantomJS unit test that looks like this:
```javascript
testUtils.renderIntoDocument( <ComponentThatWrapsANavbar />  );

// test code
```

When I run the test I get the following error:
```
RROR: 'Warning: Failed Context Types: Required context `history` was not specified in `LinkContainer`. Check the render method of `ComponentThatWrapsANavbar`.'
```

I don't see the same error when the actual page is rendered so I'm guessing it's because this component is rendered without being wrapped somewhere inside a `<Router>` component?  Is there a way I can simulate this for component testing?

**Comment by taion** [comment]
You need to wrap it in the appropriate routing context. Take a look at our tests and the React Router tests.

**Comment by thom-nic** [comment]
I'll take a look.  The actual [testing doc in react-router](https://github.com/rackt/react-router/blob/master/docs/guides/testing.md) doesn't call out anything special for testing components that use react-router.  We've been testing component that use `<Link/>`s and haven't needed to wrap them in a `<Router` for testing.  So this seems to be something specifically required for LinkContainer.  

I'm looking at LinkContainer.spec.js now and I'll try that.  Thanks for pointing me in the right direction.

**Comment by taion** [comment]
I'd accept a PR to add this logic: https://github.com/rackt/react-router/blob/master/modules/Link.js#L100-L101 to `LinkContainer`.

**Comment by thom-nic** [comment]
Ah cool so there's the cause.  Thanks I'll see if we can submit a PR.

**Comment by taion** [comment]
Fixed by #131.
